### ISSUE TITLE
Flicker in Animate Diff Pipe Outputs

### ISSUE THREAD

**Issue by AbhinavGopal** [huggingface/diffusers#7548]
### Describe the bug

For some reason, using AnimateDiffPipeline leads to some flicker in the output that is not present with the same settings in comfyui, for instance. I notice a very apparent lighting flicker between the scenes. I've attached an example here that was produced from the code snippet.
![animation](https://github.com/huggingface/diffusers/assets/43016805/80cd1e8c-9f23-486b-909e-d1c41fac466c)

### Reproduction

```
import torch
from diffusers import AnimateDiffPipeline, DDIMScheduler, MotionAdapter
from diffusers.utils import export_to_gif

# Load the motion adapter
adapter = MotionAdapter.from_pretrained("guoyww/animatediff-motion-adapter-v1-5-2", torch_dtype=torch.float16)
# load SD 1.5 based finetuned model
model_id = "SG161222/Realistic_Vision_V5.1_noVAE"
pipe = AnimateDiffPipeline.from_pretrained(model_id, motion_adapter=adapter, torch_dtype=torch.float16)
scheduler = DDIMScheduler.from_pretrained(
    model_id,
    subfolder="scheduler",
    clip_sample=False,
    timestep_spacing="linspace",
    beta_schedule="linear",
    steps_offset=1,
)
pipe.scheduler = scheduler

# enable memory savings
pipe.enable_vae_slicing()
pipe.enable_model_cpu_offload()

output = pipe(
    prompt=(
        "1girl smiling"
    ),
    negative_prompt="nsfw, bad quality, worse quality",
    num_frames=16,
    guidance_scale=7.5,
    num_inference_steps=25,
    generator=torch.Generator("cpu").manual_seed(42),
)
frames = output.frames[0]
export_to_gif(frames, "animation.gif")
```

### System Info

diffusers version: 0.28.0.dev0
Platform: Linux-4.14.322-246.539.amzn2.x86_64-x86_64-with-glibc2.31
Python version: 3.10.13
PyTorch version (GPU?): 2.2.1+cu121 (True)
Huggingface_hub version: 0.21.4
Transformers version: 4.38.2
Accelerate version: 0.28.0
xFormers version: not installed
Using GPU in script?:Yes
Using distributed or parallel set-up in script?: No

### Who can help?

@DN6 @sayakpaul @A

**Comment by bghira** [comment]
can you elaborate on your chosen scheduler settings? DDIM is known for its residual noise issues in Diffusers. Euler A seems to be more reliable.

**Comment by AbhinavGopal** [comment]
Hi @bghira, thanks for the quick response! I'm currently using DDIM. I did try Euler A, and got an improvement with the flicker. I still see some weird color patterns (like on the forehead of this gif), that I don't see on comfy. Do you have any other tips here?

As a side note, I wonder why this discrepancy exists.

![Uploading e88a80a6-30aa-4647-b0d5-2dc03c400dd1_0.gif…]()

**Comment by AbhinavGopal** [comment]
You're absolutely right! The individual frames do look significantly better :) Here's a comparison
![frame_3](https://github.com/huggingface/diffusers/assets/43016805/28dcbc58-24d0-4c58-816b-4698687bafd0)
:
![example](https://github.com/huggingface/diffusers/assets/43016805/4b0b3b16-52d4-4d3e-a4f4-1301cac979a1)
![frame_0](https://github.com/huggingface/diffusers/assets/43016805/2cc7581c-7740-4f36-a6ea-0d2fe55aa54c)
![frame_1](https://github.com/huggingface/diffusers/assets/43016805/af166bb7-0f23-4ca7-8640-8bba29b8f7a1)
![frame_2](https://github.com/huggingface/diffusers/assets/43016805/51cbf455-cc04-4c2f-aba5-9114b703a17e)


The gif shows an aberration at the start, but the images just show a brighter spot. 

Thanks for all of the help! Any idea how to bridge these gaps between diffusers and comfy? Not too sure what's different there

**Comment by yiyixuxu** [comment]
cc @DN6 should we add a note on the doc about this?

**Comment by DN6** [comment]
Thanks for the investigation! Yeah we can update the docs to reflect these results.

**Comment by DN6** [comment]
I'll also take a look into the compression issue in `export_to_gif`
